### PR TITLE
ENH: Use the "Continuous" dashboard model for master builds

### LINKED ISSUE #164: Put the Azure `master` CI builds in the *Continuous* CDash group
We can inspire from 

https://github.com/InsightSoftwareConsortium/ITK/blob/master/.circleci/config.yml#L57

CC: @blowekamp

### PR DESCRIPTION
**Pull request by thewtex**
Closes #164

### CODE REVIEW

**Comment by blowekamp** [comment]
LGTM.

There is also the "nightly-master" branch that may get move forward each night in ITK. It's not working on CircleCI either, but it would be good to do a "Nightly" model on this branch when it's updated.

**Comment by blowekamp** [comment]
@dzenanz That is because there is not CircleCI configuration for the `dashboard` branch. The [config ](https://github.com/InsightSoftwareConsortium/ITK/blob/master/.circleci/config.yml#L24)in the main branch give instructions to ignore this one, but it seems not to apply.

**Comment by thewtex** [comment]
Thanks for the reviews.

I updated the configuration to also set "Nightly" model for "nightly-master" builds. Currently, these have to be configured through Azure's web interface as opposed to the configuration file, and all Pipeline configurations should be producing Nightly builds for the `nightly-master` branch.

I assume that CircleCI is not ignoring pull requests for the `dashboard` branch because we do not have the `.circleci/config.yml` file on this branch.

**Comment by blowekamp** [comment]
> Thanks for the reviews.
> 
> I updated the configuration to also set "Nightly" model for "nightly-master" builds. Currently, these have to be configured through Azure's web interface as opposed to the configuration file, and all Pipeline configurations should be producing Nightly builds for the `nightly-master` branch.
> 
> I assume that CircleCI is not ignoring pull requests for the `dashboard` branch because we do not have the `.circleci/config.yml` file on this branch.

Correct. I looked into the web settings to see if your could set branches to ignore there... but I could not find a setting. So the `.circleci/config.yml` would need to exist in these disconnected utility branches for them to be ignored.

File: azure_dashboard.cmake [hunk 1]
```diff
@@ -45,8 +45,15 @@ if(NOT CTEST_BUILD_NAME)
...
     set(branch "-$ENV{SYSTEM_PULLREQUEST_SOURCEBRANCH}")
     set(dashboard_model "Experimental")
-  else()
+  elseif(ENV{BUILD_SOURCEBRANCHNAME} STREQUAL "master")
     set(branch "-master")
+    set(dashboard_model "Continuous")
+  elseif(ENV{BUILD_SOURCEBRANCHNAME} STREQUAL "nightly-master")
+    set(branch "-nightly-master")
+    set(dashboard_model "Nightly")
+  elseif(ENV{BUILD_SOURCEBRANCHNAME} STREQUAL "nightly-master")
```
Comment (blowekamp) [inline azure_dashboard.cmake]: Should this just be `else()`?

Comment (hjmjohnson) [reply azure_dashboard.cmake]: I think @blowekamp is correct that the last branch needs to be the catch all, and it is currently a copy of line 51.

**Comment by thewtex** [comment]
Good catch! Fixed.

**Comment by thewtex** [comment]
Merging to see what this brings on the dashboard. It may require follow-ups.

**Pull request merged**